mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -394,23 +398,25 @@ public static class LuceneMessageSearchIndexGroup extends org.apache.james.event
 
     private final MailboxId.Factory mailboxIdFactory;
     private final MessageId.Factory messageIdFactory;
+    private final LuceneIndexableDocument indexableDocument;
 
     @VisibleForTesting
     final IndexWriter writer;
     private final Directory directory;
 
     private int maxQueryResults = DEFAULT_MAX_QUERY_RESULTS;
 
-    private boolean suffixMatch = false;
+    private boolean suffixMatch = true;
```

Comment:
Why? What's the impact?

Reply:
If true, it will replace `PrefixQuery` with `WildcardQuery`.
it for `SearchCapabilities.PartialEmailMatch` 

Bellow is the document related to this variable
```
If set to true this implementation will use WildcardQuery to match suffix and prefix. This is what RFC3501 expects but is often not what the user does. It also slow things a lot if you have complex queries which use many "TEXT" arguments. If you want the implementation to behave strict like RFC3501 says, you should set this to true.
The default is false for performance reasons
```

If performance is concern, we can keep old behavior, (false)